packages/discovery-react-components/src/components/SearchFacets/SearchFacets.tsx

```diff
@@ -143,8 +146,15 @@ const SearchFacets: FC<SearchFacetsProps> = ({
       }
     }
 
-    if (searchParamsAggregationChanged || fetchState === 'init') {
+    if (
+      (searchParamsAggregationChanged || fetchState === 'init') &&
+      !isReturningFromResultPreview
```

Comment:
Is the main purpose of this just to prevent aggregations from being fetched again? To me having a state var like this feels a bit weird. Do we know which of the previous conditions were true causing these to refetch? Was `fetchState` set to `init`? or did the search params aggregation change?

Reply:
I refactored to move the `fetchState` up to `DiscoverySearch` instead. The main issue was `fetchState` being reset to `init` after document preview, yes 👍